packages/auth/lib/ConfirmationResultMFAEnroll.js

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2016-present Invertase Limited & Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this library except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+export default class ConfirmationResultMFAEnroll {
```

Comment:
This one and the SignIn class also need to be in index.d.ts exported as types I believe?

Reply:
Also - this will complicate matters a bit but in *all* ways our types and APIs must match firebase-js-sdk, the stated goal of the project is to be a drop-in replacement for people migrating from firebase-js-sdk to us so we need 100% type/API compliance including when and how errors are thrown unless native means we must do something different

Here's a general hook to the batch of auth classes that should drive the definitions here, the APIs are linked around near this:
https://firebase.google.com/docs/reference/js/firebase.auth#classes

Cases where we might deviate are for instance when we are implementing a synchronous firebase-js-sdk API but it might throw an error, and in our implementation we have to go from JS to native. The only way to throw an error in that case for us since the react-native bridge is async is for us to make the API async in our implementation so we can appropriately bubble the native error up to JS and throw it

Reply:
Yes, I agree with you. I actually started that way first and even put all the appropriate types from corresponding firebase-js-sdk here  here https://github.com/temaput/react-native-firebase/blob/implement-mfa/packages/auth/lib/index.d.ts, but then I got stuck having no Idea on how to pass opaque multifactor session https://firebase.google.com/docs/reference/ios/firebaseauth/api/reference/Classes#/c:objc(cs)FIRMultiFactorSession
across the bridge and decided to go with a shortcut.

But now, having some experience and your support I believe I could return to that path. I marked the most important questions for me to start going this way.